File: common/src/main/java/dev/number6/slack/adaptor/EnvironmentVariableSecretsConfigurationAdaptor.java
package dev.number6.slack.adaptor;

import dev.number6.slack.port.SecretsConfigurationPort;

public class EnvironmentVariableSecretsConfigurationAdaptor implements SecretsConfigurationPort {

    private static final String SLACK_TOKEN_SECRET_NAME = "SLACK_TOKEN_SECRET_NAME";

    private final String secretName = System.getenv(SLACK_TOKEN_SECRET_NAME);

    @Override
    public String getSlackTokenSecretName() {
        return secretName;
    }
}

File: slack/src/main/kotlin/dev/number6/slack/adaptor/EnvironmentVariableSecretsConfigurationAdaptor.kt
package dev.number6.slack.adaptor

import dev.number6.slack.port.SecretsConfigurationPort

internal class EnvironmentVariableSecretsConfigurationAdaptor : SecretsConfigurationPort {
    override fun getSlackTokenSecretName(): String {
        return System.getenv(SLACK_TOKEN_SECRET_NAME)
    }

    companion object {
        private const val SLACK_TOKEN_SECRET_NAME = "SLACK_TOKEN_SECRET_NAME"
    }
}